cmd/thanos/sidecar.go

```diff
@@ -371,6 +383,33 @@ func runSidecar(
 	return nil
 }
 
+func isPrometheusAgentModeEnabled(ctx context.Context, client *promclient.Client, logger log.Logger, m *promMetadata) (bool, error) {
+	var (
+		flagErr error
+		flags   promclient.Flags
+	)
+
+	if err := runutil.Retry(2*time.Second, ctx.Done(), func() error {
+		if flags, flagErr = client.ConfiguredFlags(ctx, m.promURL); flagErr != nil && flagErr != promclient.ErrFlagEndpointNotFound {
+			level.Warn(logger).Log("msg", "failed to get Prometheus flags. Is Prometheus running? Retrying", "err", flagErr)
+			return errors.Wrapf(flagErr, "fetch Prometheus flags")
+		}
+		return nil
+	}); err != nil {
+		return false, errors.Wrapf(err, "fetch Prometheus flags")
+	}
+
+	if flagErr != nil {
+		level.Warn(logger).Log("msg", "failed to check Prometheus flags, due to potentially older Prometheus. No extra validation is done.", "err", flagErr)
+		return false, nil
+	}
+	if flags.PromFeature == "agent" {
```

Comment:
I think this is a little incorrect. `enable-feature` flag on Prometheus takes in a comma-separated list of feature names. You'd need to search that via something like `strings.Contains`